### PR TITLE
feat: error highlighting

### PR DESCRIPTION
**Pull request by yndajas**
# Test

- create a commitment with just a title, then go to edit the commitment and you should see red text for questions that are missing information
- branch also fixes the pointer cursor on the edit buttons on the dashboard, so check they appear as pointers rather than the default

### CODE REVIEW

File: app/models/commitment.rb [hunk 1]
```diff
@@ -67,6 +67,14 @@ class Commitment < ApplicationRecord
     }
   ].freeze
 
+  def draft_errors
+    self.state = :live
+    valid?
+    draft_errors = errors.messages
+    self.state = :draft
```
Comment (sergiomarrocoli) [inline app/models/commitment.rb:74]: can be made a bit more concise:
```
  def draft_errors
    self.state = :live
    valid?
    self.state = :draft
    errors.messages
  end
```

File: app/javascript/components/forms/SurveyForm.vue [hunk 2]
```diff
@@ -81,6 +86,11 @@ export default {
     // call methods on library-provided events here
     model.onComplete.add(this.onComplete);
     model.onCurrentPageChanged.add(this.onCurrentPageChanged);
+    model.onUpdateQuestionCssClasses.add((survey, options) => {
+          if (this.draftErrors[options.question.name]) {
+              options.cssClasses.mainRoot += " form__question--errors";
```
Comment (sergiomarrocoli) [inline app/javascript/components/forms/SurveyForm.vue:91]: indentation

**Review by sergiomarrocoli — CHANGES_REQUESTED** [review]
works well, but the errors on the associations (countries, objectives etc) aren't showing, because the errors are under the key :countries, rather than :country_ids
might be necessary to do that manually.`

File: app/models/commitment.rb [hunk 3]
```diff
@@ -67,6 +67,19 @@ class Commitment < ApplicationRecord
     }
   ].freeze
 
+  def draft_errors
+    self.state = :live
+    valid?
+    self.state = :draft
+    errors.messages.map do |key, value|
+      if key.in?([:objectives, :managers, :countries, :actions])
```
Comment (sergiomarrocoli) [inline app/models/commitment.rb:75]: needs :threats

**Review by sergiomarrocoli — CHANGES_REQUESTED** [review]
needs :threats adding to list of associations (sorry, i must have missed it).
I think we can also change line commitment.rb:38 to ```validates :stage, inclusion: { in: STAGE_OPTIONS }, if: :live?```
there might be some issues with the cbd imports, but there seem to have been some changes to the data they want to have on commitment anyway

**Comment by sergiomarrocoli** [comment]
also need to add :implementation_year to commitment.rb:40 validiations

**Review by sergiomarrocoli — APPROVED** [review]

**Review by RainbowGiantSquid — CHANGES_REQUESTED** [review]
I'm sorry - getting a list of errors on criteria form 
image

**Pull request merged**